glue/utils/qt/autocomplete_widget.py

```diff
@@ -0,0 +1,127 @@
+# Code adapted from:
+#
+# http://rowinggolfer.blogspot.de/2010/08/qtextedit-with-autocompletion-using.html
+#
+# and based on:
+#
+# http://qt-project.org/doc/qt-4.8/tools-customcompleter.html
+
+from ...external.qt import QtGui, QtCore
+
+
+__all__ = ["CompletionTextEdit"]
+
+try:  # Python 2
+
+    QString = QtCore.QString
+
+except AttributeError:  # Python 3
+
+    class QString(str):
+        def isEmpty(self):
+            return len(self) > 0
+        def length(self):
+            return len(self)
+        def contains(self, sub):
+            return sub in self
+        def right(self, n):
+            try:
+                return self[-n:]
+            except IndexError:
+                return self
+
+
+class CompletionTextEdit(QtGui.QTextEdit):
+
+    def __init__(self, parent=None):
+
+        super(CompletionTextEdit, self).__init__(parent)
+
+        self.setMinimumWidth(400)
+        self.completer = None
+        self.word_list = None
+
+        self.moveCursor(QtGui.QTextCursor.End)
+
+    def set_word_list(self, word_list):
+        self.word_list = word_list
+        self.set_completer(QtGui.QCompleter(word_list))
+
+    def set_completer(self, completer):
+
+        if self.completer:
+            self.disconnect(self.completer, 0, self, 0)
+        if not completer:
+            return
+
+        completer.setWidget(self)
+        completer.setCompletionMode(QtGui.QCompleter.PopupCompletion)
+        completer.setCaseSensitivity(QtCore.Qt.CaseInsensitive)
+        self.completer = completer
+        self.connect(self.completer, QtCore.SIGNAL("activated(const QString&)"), self.insert_completion)
```

Comment:
Any reason you didn't use self.completer.activated.connect(self.insert_completion)? That's the style used elsewhere in the code